source/jormungandr/jormungandr/interfaces/v1/fields.py

```diff
@@ -843,18 +843,4 @@ class DisruptionsField(fields.Raw):
     """
 
     def output(self, key, obj):
-        all_impacts = {}
-
-        def get_all_impacts(_, val):
-            if not hasattr(val, 'impacts'):
-                return
-            impacts = val.impacts
-            if not impacts or not hasattr(impacts[0], 'uri'):
-                return
-
-            for impact in impacts:
-                all_impacts[impact.uri] = impact
-
-        utils.walk_protobuf(obj, get_all_impacts)
-
-        return [marshal(d, disruption_marshaller, display_null=False) for d in all_impacts.values()]
+        return [marshal(d, disruption_marshaller, display_null=False) for d in obj.impacts]
```

Comment:
I think you can replace DisruptionsField with NonNullList(NonNullNested(disruption_marshaller))